lib/auth/native/native.go

```diff
@@ -75,31 +75,31 @@ func generateKeyPairImpl() ([]byte, []byte, error) {
 	return privPem, pubBytes, nil
 }
 
-func replenishKeys() {
-	// Mark the task as stopped.
-	defer atomic.StoreInt32(&precomputeTaskStarted, 0)
-
+func precomputeKeys() {
+	const backoff = time.Second * 30
 	for {
 		priv, pub, err := generateKeyPairImpl()
 		if err != nil {
-			log.Errorf("Failed to generate key pair: %v", err)
-			return
+			log.WithError(err).Errorf("Failed to precompute key pair, retrying in %s (this might be a bug).", backoff)
+			time.Sleep(backoff)
 		}
 
 		precomputedKeys <- keyPair{priv, pub}
 	}
 }
 
-// GenerateKeyPair returns fresh priv/pub keypair, takes about 300ms to execute in a worst case.
-// This will in most cases pull from a precomputed cache of ready to use keys.
-func GenerateKeyPair() ([]byte, []byte, error) {
-	// Start the background task to replenish the queue of precomputed keys.
-	// This is only started once this function is called to avoid starting the task
-	// just by pulling in this package.
-	if atomic.SwapInt32(&precomputeTaskStarted, 1) == 0 {
-		go replenishKeys()
+// PrecomputeKeys sets this package into a mode where a small backlog of keys are
+// computed in advance.  This should only be enabled if large spikes in key computation
+// are expected (e.g. in auth/proxy services).  Safe to double-call.
+func PrecomputeKeys() {
+	if !precomputeTaskStarted.Swap(true) {
```

Comment:
Can we use `sync.Once` here? We're never stopping the task after all, and would get rid of the atomic bool from go.uber.com

Reply:
@rosstimothy I changed it to warn and backoff instead of exit.  Old behavior restarted it on each call to `native.GenerateKeyPair`, so this is equivalent to old behavior since we now only start it once.